### PR TITLE
chore(deps): update leafygreen packages to latest

### PR DESCRIPTION
**Pull request by mabaasit**
Currently draft because i need to check update for `@emotion/css`

_And_ also popover v12 changes.

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-components/src/hooks/use-virtual-grid.test.tsx [hunk 1]
```diff
@@ -38,6 +38,7 @@ const TestGrid: React.FunctionComponent<{
             const idx = row * colCount + col;
             return (
               <div
+                key={idx}
```
Comment (mabaasit) [inline packages/compass-components/src/hooks/use-virtual-grid.test.tsx:41]: small drive-by

File: packages/compass-components/src/hooks/use-confirmation.tsx [hunk 2]
```diff
@@ -176,7 +175,6 @@ export const ConfirmationModalArea: React.FC = ({ children }) => {
           className: confirmationProps.hideCancelButton
             ? hideButtonStyles
             : undefined,
-          children: confirmationProps.cancelButtonText ?? 'Cancel',
```
Comment (mabaasit) [inline packages/compass-components/src/hooks/use-confirmation.tsx:179]: button text is always Cancel and LG removed this option. https://www.mongodb.design/component/confirmation-modal/code-docs

File: .github/workflows/update-leafygreen-packages.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Install npm@10.2.4
+        run: |
+          npm install -g npm@10.2.4
+      - name: Install Dependencies
+        run: |
+          npm -v
+          npm ci
+      - name: Bump packages
+        run: |
+          npm add $(cat packages/compass-components/package.json | jq -r '.dependencies | keys[]' | grep "@leafygreen-ui" | sed s/$/@latest/ | xargs)  --workspace=@mongodb-js/compass-components
```
Comment (mabaasit) [inline .github/workflows/update-leafygreen-packages.yaml:41]: Copied from https://github.com/10gen/compass-data-explorer/#publishing-packages

Comment (mabaasit) [reply .github/workflows/update-leafygreen-packages.yaml:41]: now i am thinking about `jq` - would it be available on CI 🤔 ?

Comment (mabaasit) [reply .github/workflows/update-leafygreen-packages.yaml:41]: and its available there: https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#tools

File: .github/workflows/update-leafygreen-packages.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+name: Update leafygreen packages
```
Comment (mabaasit) [inline .github/workflows/update-leafygreen-packages.yaml:1]: this whole file is a copy of `update-electron` with lg changes

File: packages/compass-global-writes/src/store/reducer.ts [hunk 5]
```diff
@@ -625,7 +625,8 @@ export const createShardKey = (
 
 // Exporting this for test only to stub it and set
 // its value. This enables to test cancelShardingaction.
-export const showConfirmation = showConfirmationModal;
+export const showConfirmation: typeof showConfirmationModal =
```
Comment (kraenhansen) [inline packages/compass-global-writes/src/store/reducer.ts:628]: I'm curious as to why this is needed? 🤔 It seems redundant to me.
Same with the change in `packages/compass-indexes/src/modules/regular-indexes.ts`

Comment (mabaasit) [reply packages/compass-global-writes/src/store/reducer.ts:628]: ts started to complain about this
`The inferred type of 'showConfirmation' cannot be named without a reference to '@mongodb-js/compass-components/node_modules/@leafygreen-ui/confirmation-modal'. This is likely not portable. A type annotation is necessary.ts(2742)`.

I'll close this PR as I split the removal of combobox from this.
